packages/account/src/Components/address-details/address-details.tsx

```diff
@@ -7,176 +16,183 @@ import {
+        goToNextStep,
+        goToPreviousStep,
+        validate,
+        onSubmit,
+        is_svg,
+        is_mf,
+        is_gb_residence,
+        onSubmitEnabledChange,
+        selected_step_ref,
+        disabled_items,
+        has_real_account,
+        ...props
+    }: TAddressDetails) => {
+        const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
+        const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
 
-const AddressDetails = ({
-    states_list,
-    getCurrentStep,
-    onSave,
-    onCancel,
-    goToNextStep,
-    goToPreviousStep,
-    validate,
-    onSubmit,
-    is_svg,
-    is_mf,
-    is_gb_residence,
-    onSubmitEnabledChange,
-    selected_step_ref,
-    disabled_items,
-    has_real_account,
-    ...props
-}: TAddressDetails) => {
-    const { is_appstore } = React.useContext(PlatformContext);
-    const [has_fetched_states_list, setHasFetchedStatesList] = React.useState(false);
-    const [address_state_to_display, setAddressStateToDisplay] = React.useState('');
+        const { client, ui } = useStore();
 
-    React.useEffect(() => {
-        const { cancel, promise } = makeCancellablePromise(props.fetchStatesList());
-        promise.then(() => {
-            setHasFetchedStatesList(true);
-            if (props.value.address_state) {
-                setAddressStateToDisplay(getLocation(states_list, props.value.address_state, 'text'));
-            }
-        });
-        return () => {
-            setHasFetchedStatesList(false);
-            cancel();
-        };
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+        const { states_list, fetchStatesList } = client;
+        const { is_desktop, is_mobile } = ui;
 
-    const is_submit_disabled_ref = React.useRef<boolean | undefined>(true);
+        React.useEffect(() => {
+            const { cancel, promise } = makeCancellablePromise(fetchStatesList());
+            promise.then(() => {
+                setHasFetchedStatesList(true);
+                if (props.value.address_state) {
+                    setAddressStateToDisplay(getLocation(states_list, props.value.address_state, 'text'));
+                }
+            });
+            return () => {
+                setHasFetchedStatesList(false);
+                cancel();
+            };
+        }, []);
 
-    const isSubmitDisabled = (errors?: { [key: string]: string } | FormikValues) => {
-        return selected_step_ref?.current?.isSubmitting || (errors && Object.keys(errors).length > 0);
-    };
+        const is_submit_disabled_ref = React.useRef<boolean | undefined>(true);
 
-    const checkSubmitStatus = (errors?: { [key: string]: string }) => {
-        const is_submit_disabled = isSubmitDisabled(errors);
+        const isSubmitDisabled = (errors?: FormikErrors<TAddressDetailFormProps>) => {
+            return selected_step_ref?.current?.isSubmitting || (errors && Object.keys(errors).length > 0);
```

Comment:
the flag `isValid` from formik returns true if the `errors` object is empty 
Ref: https://formik.org/docs/api/formik#isvalid-boolean

Reply:
Refactored the code. The function was not required